### PR TITLE
Update Example and Templates

### LINKED ISSUE #5942: Update Expo template to use the local Expo CLI
Our Expo template uses the global `expo-cli` that is not supported anymore (and it does not support Node +17). We can update it (along with the README) to use the local `@expo/cli`.

See:
* Blog post: [The New Expo CLI](https://blog.expo.dev/the-new-expo-cli-f4250d8e3421)
* @expo/cli: [README](https://github.com/expo/expo/blob/main/packages/%40expo/cli/README.md)

### PR DESCRIPTION
**Pull request by takameyer**
## What, How & Why?
* Fix small bug in error for auth hooks
* Update dependencies
* Update React Native and Expo in Templates

This closes #5942

We should wait for [this](https://github.com/expo/expo-cli/pull/4750) to be merged before releasing the expo template.

## ☑️ ToDos

* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

File: templates/expo-template/README.md
Comment (elle-j) [inline templates/expo-template/README.md:1]: Should we mention anything about npm v9 here?

File: templates/react-native-template/template/README.md
Comment (elle-j) [inline templates/react-native-template/template/README.md:1]: I noticed that a README inside the `react-native-template/template/` directory didn't exist before. Is this README meant to coexist with [react-native-template/README.md](https://github.com/realm/realm-js/blob/andrew/update-templates/templates/react-native-template/README.md)?

File: templates/react-native-template/template/android/gradle/wrapper/gradle-wrapper.jar
Comment (elle-j) [inline templates/react-native-template/template/android/gradle/wrapper/gradle-wrapper.jar:1]: Note: Binary got uploaded.

File: templates/react-native-template/template/index.js [hunk 1]
```diff
@@ -17,4 +17,6 @@ const App = () =>
     <AppWrapperNonSync />
   );
 
+export default App;
+
 AppRegistry.registerComponent(appName, () => App);
```
Comment (elle-j) [inline templates/react-native-template/template/index.js:22]: Is there something importing `App`? Seemed to only be passed to `AppRegistry` before.

File: templates/react-native-template/template/package.json [hunk 2]
```diff
@@ -10,32 +10,31 @@
...
   },
   "dependencies": {
-    "@realm/react": "^0.5.1",
+    "@realm/react": "^0.6.0",
     "react": "18.2.0",
-    "react-native": "0.71.11",
+    "react-native": "0.72.4",
     "react-native-get-random-values": "^1.9.0",
-    "realm": "11.10.1"
+    "realm": "12.0.0"
```
Comment (elle-j) [inline templates/react-native-template/template/package.json:17]: 🍾

File: templates/react-native-template/template/sync.config.js
Comment (elle-j) [inline templates/react-native-template/template/sync.config.js:1]: Might have missed it, but did the contents of this file get moved somewhere else?

**Review by elle-j — APPROVED** [review]
LGTM, with some minor questions 👍

File: templates/react-native-template/template/index.js (hunk 1 above)
Comment (takameyer) [reply templates/react-native-template/template/index.js:22]: The Jest test

File: templates/react-native-template/template/sync.config.js
Comment (takameyer) [reply templates/react-native-template/template/sync.config.js:1]: Good catch.  I generally just create a new react native project to upgrade the templates and copy the files in by hand.  I had forgotten this one.  Perhaps one day I'll actually get this automated in a sane way.

File: templates/react-native-template/template/README.md
Comment (takameyer) [reply templates/react-native-template/template/README.md:1]: Yes, it's fine.  React Native added a readme to their base template in version 72.  I like to keep the templates as near to the React Native default.  But now that you have pointed this out, I will added pieces of our template README into this one to explain sync

File: .vscode/launch.json [hunk 3]
```diff
@@ -4,6 +4,7 @@
     // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
     "version": "0.2.0",
     "configurations": [
+
```
[the remark this answers is not in the document]
Comment (takameyer) [reply .vscode/launch.json:7]: Accidental modification, yes

File: templates/react-native-template/template/android/gradle/wrapper/gradle-wrapper.jar
Comment (takameyer) [reply templates/react-native-template/template/android/gradle/wrapper/gradle-wrapper.jar:1]: That's fine, it's coming from the latest React Native

File: templates/react-native-template/template/sync.config.js
Comment (elle-j) [reply templates/react-native-template/template/sync.config.js:1]: I agree, it's often easier to just create a new project 👍

Comment (takameyer) [reply templates/react-native-template/template/sync.config.js:1]: Expo is the easiest.  It actually can upgrade automatically.

**Pull request merged**